packages/js/customer-effort-score/src/components/customer-effort-score-tracks/index.js

```diff
@@ -58,7 +58,15 @@ function _CustomerEffortScoreTracks( {
 	updateOptions,
 	createNotice,
 } ) {
-	const [ modalShown, setModalShown ] = useState( false );
+	useEffect( () => {
+		if ( cesShownForActions && ! cesShownForActions.includes( action ) )
```

Comment:
This logic is wrong, it should be if `cesShownForActions` does not exist or `! cesShownForActions.includes( action )`
Also calling it on initial render seems to break this, as the component is not done `resolving` its data yet, so `cesShownForActions` is always `undefined`, meaning that the previous value will always be overwritten.

I think a better option would be to add this logic in the `onNoticeShown` callback, this way we always know the actual component has finished resolving and tracking is also enabled.
We can still add the `if ( ! cesShownForActions || ! cesShownForActions.includes( action ) )` check there to make sure we don't add duplicates.

Reply:
Thank you @louwie17 for reviewing this.
I forgot to push my latest changes 🤦🏻‍♂️. Apologies for that 🙏
I fixed it in the commit [078074e](https://github.com/woocommerce/woocommerce/pull/43164/commits/078074ed68b4e93510033b78533e8e4eae099ff1)